packages/key-management/src/TrezorKeyAgent.ts

```diff
@@ -52,11 +61,12 @@ export class TrezorKeyAgent extends KeyAgentBase {
     silentMode = false,
     lazyLoad = false
   }: TrezorConfig): Promise<boolean> {
+    const trezorConnect = getTrezorConnectInstance(communicationType);
```

Comment:
I don't think they have to be static, but one thing to have in mind is that connection to the device probably has to be initiated synchronously on user gesture, so a separate function to connect can be convenient. I'm sure we can improve this API, but need to review how it's being used in Lace.

Reply:
@marcin-mazurek 
1. yeah instance is always initialized with `new` but in this case, I had a dilemma and I rather used an instance keyword to say what package you will use instead of saying e.g. `getTrezorConnectPackage` or similar. 
2. `static` methods: That is an idea from the very first implementation and it is quite useful when you want to instantiate ledger or trezor communication on your own and without instantiating TrezorKeyAgent. For example, you just want to check the device connection or you want to create device connection and pass it to `createWithDevice` [here](https://github.com/input-output-hk/cardano-js-sdk/blob/12d8422fadb3ced2d0ffc1d00c611d3728713870/packages/hardware-ledger/src/LedgerKeyAgent.ts#L266C53-L266C69) . In that case, you are able to communicate with a device on your own without KeyAgent instances and without installing ledger/trezor packages in the app. Let's say that you have more open hands in that way

Reply:
@tomislavhoracek 

1. Two proposals not using `instance` keyword: how about `getTrezorConnect` or `getTrezorConnectByCommunicationType`?
2. OK - thanks for the explanation, considering the fact that Ledger is also done this way and any changes would result in a lot of breaking changes, it'd probably make sense to stick to the current approach

Reply:
We discussed this in a call and we decided to use `getTrezorConnect` instead of `getTrezorConnectInstance`
cc @rhyslbw @mkazlauskas